src/bin/server/attestations/mod.rs

```diff
@@ -0,0 +1,557 @@
+pub struct PriceValueEntry {
+    pub validator_public_key: String,
+    pub value: i64,
+    pub slot_number: i64,
+    pub signature: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PriceIntervalEntry {
+    pub validator_public_key: String,
+    pub value: i64,
+    pub slot_number: i64,
+    pub signature: String,
+    pub interval_size: i64,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct AggregatePriceIntervalEntry {
+    pub value: i64,
+    pub slot_number: i64,
+    pub aggregate_signature: String,
+    pub aggregate_public_key: String,
+    pub interval_size: i64,
+    pub num_validators: i64,
+}
+
+#[derive(Clone, Debug, Encode, Decode, Serialize, Deserialize)]
+pub struct Price {
+    pub value: u64, // TODO: Check if we need to add further info here such as timestamp
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct OracleMessage {
+    pub value_message: SignedPriceValueMessage,
+    pub interval_inclusion_messages: Vec<SignedIntervalInclusionMessage>,
+    pub validator_public_key: PublicKey,
+}
+
+#[derive(Clone, Debug, Decode, Encode, Serialize, Deserialize)]
+pub struct PriceValueMessage {
+    pub price: Price,
+    pub slot_number: u64,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SignedPriceValueMessage {
+    pub message: PriceValueMessage,
+    pub signature: Signature,
+}
+
+#[derive(Clone, Debug, Decode, Encode, Serialize, Deserialize)]
+pub struct IntervalInclusionMessage {
+    pub value: u64,
+    pub interval_size: u64,
+    pub slot_number: u64,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SignedIntervalInclusionMessage {
+    pub message: IntervalInclusionMessage,
+    pub signature: Signature,
+}
+
+pub async fn get_price_value_attestations(
+    State(state): State<Arc<AppState>>,
+) -> Json<Vec<PriceValueEntry>> {
+    let db_pool = &state.db_pool;
+    let entries: Vec<PriceValueEntry> = sqlx::query!(
+        "
+        SELECT
+            validator_public_key,
+            value,
+            slot_number,
+            signature
+        FROM
+            price_value_attestations;
+        "
+    )
+    .fetch_all(db_pool)
+    .await
+    .unwrap()
+    .into_iter()
+    .map(|row| PriceValueEntry {
+        validator_public_key: row.validator_public_key,
+        value: row.value,
+        slot_number: row.slot_number,
+        signature: row.signature,
+    })
+    .collect();
+    Json(entries)
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct PriceAggregateQueryParams {
+    slot_number: Option<i64>,
+    interval_size: Option<i64>,
+}
+
+pub async fn get_price_aggregate(
+    Query(query): Query<PriceAggregateQueryParams>,
+    State(state): State<Arc<AppState>>,
+) -> Result<Json<AggregatePriceIntervalEntry>, axum::http::StatusCode> {
+    let db_pool = &state.db_pool;
+    tracing::debug!("Query: {:?}", query);
+    // TODO: Improve error / none handling - remove unwrap
```

Comment:
yea hah, I'm fine leaving the todo but this feels like the level where all errors are either logged and replied 500, or unwrapped (perhaps with an `expect` to clarify what assumption was broken that means we need to crash the whole server).

So for query params I'd be curious if there was some way to make it mandatory or short-circuit to 401. Then when we have what we need, try to get the slot number which again, either succeeds, or short-circuits with the mapped error. Now, if `get_slot_number` hits a DB issue, where we should probably panic, it responds 500 (ok too, but panic might be more safe because we'd restart), yet when it cannot find the slot, which is probably acceptable, we panic.